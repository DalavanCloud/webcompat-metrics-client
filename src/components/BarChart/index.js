import React from "react";
import { Bar, Chart } from "react-chartjs-2";

import {
  propTypes,
  defaultProps,
  hydrateData,
} from "../../propTypes/ChartPropTypes.js";
import { Container } from "../Chart";

class BarChart extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    Chart.pluginService.register({
      afterDraw: (chart, easing) => {
        const ctx = chart.ctx;
        chart.data.datasets.forEach((dataset, i) => {
          const meta = chart.getDatasetMeta(i);
          if (!meta.hidden) {
            meta.data.forEach((element, index) => {
              // Draw the text in black, with the specified font
              ctx.fillStyle = "rgb(0, 0, 0)";
              const fontSize = 16;
              const fontStyle = "normal";
              const fontFamily = "Helvetica Neue";
              ctx.font = Chart.helpers.fontString(
                fontSize,
                fontStyle,
                fontFamily,
              );
              // Just naively convert to string for now
              const dataString = dataset.data[index].toString();
              // Make sure alignment settings are correct
              ctx.textAlign = "center";
              ctx.textBaseline = "middle";
              const padding = 5;
              const position = element.tooltipPosition();
              ctx.fillText(
                dataString,
                position.x,
                position.y - fontSize / 2 - padding,
              );
            });
          }
        });
      },
    });
  }

  render() {
    const data = {
      ...hydrateData(this.props),
    };
    return (
      <Container title={this.props.title} subtitle={this.props.subtitle}>
        <Bar
          data={data}
          options={this.props.options}
          legend={this.props.legend}
        />
      </Container>
    );
  }
}

BarChart.propTypes = {
  ...propTypes,
};

BarChart.defaultProps = {
  ...defaultProps,
};

export default BarChart;
