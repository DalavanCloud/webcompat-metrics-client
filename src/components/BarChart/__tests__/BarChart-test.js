/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";

import BarChart from "..";

jest.mock("react-chartjs-2", () => ({ Bar: () => <div>Chart</div> }));

it("renders BarChart default correctly", () => {
  const tree = renderer
    .create(
      <BarChart
        title={"Title BarCHart"}
        labels={["1", "2"]}
        data={["1000", "2000"]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
