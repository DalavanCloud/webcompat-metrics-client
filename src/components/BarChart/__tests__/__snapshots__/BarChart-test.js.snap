// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BarChart default correctly 1`] = `
<article
  className="container"
>
  <div
    className="wrapper"
  >
    <h3
      className="title"
    >
      Title BarCHart
    </h3>
    <div>
      <div>
        Chart
      </div>
    </div>
  </div>
</article>
`;
